segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/hdfs/HDFSStorage.java

```diff
@@ -136,52 +139,187 @@ public void initialize(long epoch) {
             conf.set("dfs.client.block.write.replace-datanode-on-failure.policy", "NEVER");
         }
 
-        this.context = new FileSystemOperation.OperationContext(epoch, openFileSystem(conf), this.config);
+        this.context = new OperationContext(epoch, openFileSystem(conf), this.config);
         log.info("Initialized (HDFSHost = '{}', Epoch = {}).", this.config.getHdfsHostURL(), epoch);
     }
 
-    protected FileSystem openFileSystem(Configuration conf) throws IOException {
-        return FileSystem.get(conf);
+    @Override
+    public SegmentProperties getStreamSegmentInfo(String streamSegmentName) throws StreamSegmentException {
+        long traceId = LoggerHelpers.traceEnter(log, "getStreamSegmentInfo", streamSegmentName);
+        try {
+            return Retry.withExpBackoff( 1, 5, MAX_ATTEMPT_COUNT)
+                        .retryingOn(FileNotFoundException.class)
+                        .throwingOn(IOException.class)
+                        .run(() -> {
+                            long length = 0;
+                            boolean isSealed = false;
+                            FileStatus last = null;
+                            SegmentProperties result = null;
+                            last = findStatusForSegment(streamSegmentName, true);
+                            isSealed = isSealed(last.getPath());
+                            result = StreamSegmentInformation.builder().name(streamSegmentName).length(last.getLen()).sealed(isSealed).build();
+                            LoggerHelpers.traceLeave(log, "getStreamSegmentInfo", traceId, streamSegmentName, result);
+                            return result;
+                        });
+        } catch (IOException e) {
+            HDFSExceptionHelpers.throwException(streamSegmentName, e);
+        } catch (RetriesExhaustedException e) {
+            HDFSExceptionHelpers.throwException(streamSegmentName, e.getCause());
+        }
+        return null;
     }
 
     @Override
-    public SegmentProperties create(String streamSegmentName) throws StreamSegmentException {
-        return call(new CreateOperation(streamSegmentName, this.context));
+    public boolean exists(String streamSegmentName) {
+        long traceId = LoggerHelpers.traceEnter(log, "exists", streamSegmentName);
+        FileStatus status;
+        try {
+            status = findStatusForSegment(streamSegmentName, false);
+        } catch (IOException e) {
+            return false;
```

Comment:
Please add an explanation why you return `false` upon an exception.

Reply:
Fixed.